This upstream accepted patch fixes a bug where the live images don't work when
booted from grub loopback (void-linux/void-mklive#294).

From d880d62f5f81d7ec69555f5deb60694fdb693c01 Mon Sep 17 00:00:00 2001
From: 0x5c <dev@0x5c.io>
Date: Thu, 9 Feb 2023 05:26:30 -0500
Subject: [PATCH] fix: make iso-scan trigger udev events

Fixes #2183
---
 modules.d/90dmsquash-live/iso-scan.sh | 1 +
 1 file changed, 1 insertion(+)

diff --git a/modules.d/90dmsquash-live/iso-scan.sh b/modules.d/90dmsquash-live/iso-scan.sh
index 886d23033..fa06b33cd 100755
--- a/modules.d/90dmsquash-live/iso-scan.sh
+++ b/modules.d/90dmsquash-live/iso-scan.sh
@@ -22,6 +22,7 @@ do_iso_scan() {
         mount -t auto -o ro "$dev" "/run/initramfs/isoscan" || continue
         if [ -f "/run/initramfs/isoscan/$isofile" ]; then
             losetup -f "/run/initramfs/isoscan/$isofile"
+            udevadm trigger --action=add > /dev/null 2>&1
             ln -s "$dev" /run/initramfs/isoscandev
             rm -f -- "$job"
             exit 0
